src/Agent.Worker/Build/SvnCommandManager.cs

```diff
@@ -0,0 +1,828 @@
+        /// <param name="rootPath"></param>
+        /// <returns></returns>
+        string ResolveServerPath(string serverPath, string rootPath);
+    }
+
+    public class SvnCommandManager : AgentService, ISvnCommandManager
+    {
+        public void Init(
+            IExecutionContext context,
+            ServiceEndpoint endpoint,
+            CancellationToken cancellationToken)
+        {
+            // Validation.
+            ArgUtil.NotNull(context, nameof(context));
+            ArgUtil.NotNull(endpoint, nameof(endpoint));
+            ArgUtil.NotNull(cancellationToken, nameof(cancellationToken));
+
+            ArgUtil.NotNull(endpoint.Url, nameof(endpoint.Url));
+            ArgUtil.Equal(true, endpoint.Url.IsAbsoluteUri, nameof(endpoint.Url.IsAbsoluteUri));
+            ArgUtil.NotNull(endpoint.Data, nameof(endpoint.Data));
+
+            ArgUtil.NotNull(endpoint.Authorization, nameof(endpoint.Authorization));
+            ArgUtil.NotNull(endpoint.Authorization.Parameters, nameof(endpoint.Authorization.Parameters));
+            ArgUtil.Equal(EndpointAuthorizationSchemes.UsernamePassword, endpoint.Authorization.Scheme, nameof(endpoint.Authorization.Scheme));
+
+            _context = context;
+            _endpoint = endpoint;
+            _cancellationToken = cancellationToken;
+
+            //find git in %Path%
+            IWhichUtil whichTool = HostContext.GetService<IWhichUtil>();
+            string svnPath = whichTool.Which("svn");
+
+            if (string.IsNullOrEmpty(svnPath))
+            {
+                throw new Exception(StringUtil.Loc("SvnNotInstalled"));
+            }
+            else
+            {
+                _context.Debug($"Found svn installation path: {svnPath}.");
+                _svn = svnPath;
+            }
+
+            // External providers may need basic auth or tokens
+            endpoint.Authorization.Parameters.TryGetValue(EndpointAuthorizationParameters.Username, out _username);
+            endpoint.Authorization.Parameters.TryGetValue(EndpointAuthorizationParameters.Password, out _password);
+
+            _acceptUntrusted = endpoint.Data.ContainsKey(/* WellKnownEndpointData.SvnAcceptUntrustedCertificates */ "acceptUntrustedCerts") &&
+            StringUtil.ConvertToBoolean(endpoint.Data[/* WellKnownEndpointData.SvnAcceptUntrustedCertificates */ "acceptUntrustedCerts"], defaultValue: false);
+        }
+
+        public async Task<string> UpdateWorkspace(
+            string rootPath,
+            Dictionary<string, SvnMappingDetails> distinctMappings,
+            bool cleanRepository,
+            string sourceBranch,
+            string revision)
+        {
+            if (cleanRepository)
+            {
+                RecreateDirectory(rootPath);
+            }
+
+            Dictionary<string, Uri> oldMappings = await GetOldMappings(rootPath);
+            if (_context.Variables.System_Debug.HasValue && _context.Variables.System_Debug.Value)
```

Comment:
You don't need check System_debug, when you call context.debug, it will check it for you.